src/SldStyleParser.ts

```diff
@@ -388,43 +371,83 @@ export class SldStyleParser implements StyleParser<string> {
    * @return {string} The name to be used for the GeoStyler Style Style
    */
   getStyleNameFromSldObject(sldObject: any): string {
-    const userStyleTitle = _get(sldObject, 'StyledLayerDescriptor.NamedLayer[0].UserStyle[0].Title[0]._');
-    const namedLayerName = _get(sldObject, 'StyledLayerDescriptor.NamedLayer[0].Name[0]._');
+    const userStyleTitle = get(sldObject, 'StyledLayerDescriptor.NamedLayer[0].UserStyle.Title.#text');
+    const namedLayerName = get(sldObject, 'StyledLayerDescriptor.NamedLayer.Name.#text');
     return userStyleTitle ? userStyleTitle
       : namedLayerName ? namedLayerName : '';
   }
 
   /**
-   * Creates a GeoStyler-Style StrMatchesFunctionFilterr from a SLD strMatches Function.
+   * Get the GeoStyler-Style Filter from an SLD Rule.
    *
-   * @param {object} sldFilter The SLD Filter
-   * @return {Filter} The GeoStyler-Style FunctionFilter
+   * Currently only supports one Filter per Rule.
+   *
+   * @param sldRule The SLD Rule
+   * @return The GeoStyler-Style Filter
    */
-  getStrMatchesFunctionFilterFromSldFilter(sldFilter: any): StrMatchesFunctionFilter {
-    const propertyName = _get(sldFilter, 'Function[0].PropertyName[0]._');
-    const literal = _get(sldFilter, 'Function[0].Literal[0]._');
-    const regex = new RegExp(literal);
-    return [
-      'FN_strMatches',
-      propertyName,
-      regex
-    ];
+  getFilterFromRule(sldRule: any[]): GsFilter | undefined {
+    const sldFilter = get(sldRule, 'Filter');
+    if (!sldFilter || sldFilter.length === 0) {
+      return;
+    }
+    const operator = Object.keys(sldFilter[0])?.[0];
+    if (!operator) {
+      return;
+    }
+    const filter = this.getFilterFromOperatorAndComparison(operator, sldFilter);
+    return filter;
   }
 
   /**
-   * Creates a GeoStyler-Style FunctionFilter from a SLD Function.
+   * Get the GeoStyler-Style ScaleDenominator from an SLD Rule.
```

Comment:
```suggestion
   * Get the GeoStyler-Style ScaleDenominator from a SLD Rule.
```

Probably also elsewhere